### PR TITLE
Remove `Regex::PCRE2#finalize` redefinition

### PR DESCRIPTION
**Pull request by HertzDevil**
This redefinition made very simple snippets like below run out of memory eventually:

```crystal
while true
  Regex.new("")
end
```

It might be related to #13306 and some other recent memory blow-ups too.

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]
Great find 🚀 🙈 
But generating the docs still stalls: https://github.com/crystal-lang/crystal/actions/runs/4683438004/jobs/8298519048?pr=13309
So it doesn't seem to fix that.

File: src/regex/pcre2.cr [hunk 1]
```diff
@@ -279,12 +276,6 @@ module Regex::PCRE2
     match_data
   end
 
-  def self.config(what, type : T.class) : T forall T
-    value = uninitialized T
-    LibPCRE2.config(what, pointerof(value))
-    value
-  end
-
   module MatchData
```
Comment (beta-ziliani) [inline src/regex/pcre2.cr:279]: What's the reason for this removal? It isn't used? Is there a relation with #13307?

Comment (HertzDevil) [reply src/regex/pcre2.cr:279]: It's unused, `LibPCRE2.config` is directly called

**Comment by HertzDevil** [comment]
d60fe6c: `LibPCRE2.code_free` can be called freely even in Crystal's interpreter, because it is a regular lib fun, not an extern `Proc`-typed variable, unlike `LibPCRE.free` (#12495).

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**